artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/LinkedListImpl.java

```diff
@@ -203,20 +223,31 @@ public void addSorted(E e) {
          // This would be an optimization for our usage.
          // avoiding scanning the entire List just to add at the end, so we compare the end first.
          if (comparator.compare(tail.val(), e) >= 0) {
+            logger.trace("addTail as compare is after tail {}", e);
             addTail(e);
             return;
          }
 
-         Node<E> fetching = head.next;
-         while (fetching.next != null) {
-            int compareNext = comparator.compare(fetching.next.val(), e);
-            if (compareNext <= 0) {
-               addAfter(fetching, e);
-               return;
+         if (lastAdd != null) { // as an optimization we check against the last add rather than always scan.
+            if (lastAdd.prev != null && lastAdd.prev.val() != null) {
+               if (comparator.compare(lastAdd.prev.val(), e) > 0 && comparator.compare(lastAdd.val(), e) < 0) {
+                  logger.trace("addABefore last element {}", e);
+                  addAfter(lastAdd.prev, e);
+                  return;
+               }
+            }
+            if (lastAdd.next != null && lastAdd.next.val() != null) {
+               if (comparator.compare(lastAdd.val(), e) > 0 && comparator.compare(lastAdd.next.val(), e) < 0) {
+                  logger.trace("addAfter last element {}", e);
+                  addAfter(lastAdd, e);
+                  return;
+               }
             }
-            fetching = fetching.next;
          }
 
+         if (addSortedScan(e))
+            return;
```

Comment:
Can we spend that 1 line on braces? :) More consistent and  less likely to be messed up later.

Reply:
lol

I didn't think about it.. just right click, refactor as a new method.. automatic change :)